homeassistant/components/iotawatt/sensor.py

```diff
@@ -151,30 +160,37 @@ def __init__(
         self,
         coordinator,
         key,
-        mac_address,
-        name,
         entity_description: IotaWattSensorEntityDescription,
     ):
         """Initialize the sensor."""
         super().__init__(coordinator=coordinator)
 
         self._key = key
         data = self._sensor_data
+        self._accumulating = data.getUnit() == "WattHours" and not data.getFromStart()
```

Comment:
These look like a new type of sensor (lots of `if self._accumulating` branches). It should probably get its own class and share the existing sensors with a new base class.

Reply:
great idea, indeed makes the code much more readable and remove unnecessary tests